### ISSUE TITLE
adding bike/ped infrastructure to SharedStreets RoadClass hierarchy

### ISSUE THREAD

**Issue by kpwebb** [sharedstreets/sharedstreets-ref-system#24]
Currently SharedStreets includes only public car-routable paths on public streets. 

We're seeing increasing interest in including bike/ped only infrastructure, and potential downstream data consumer applications that can close or place restrictions on bike/ped only facilities . 

To support these applications we need to include references for these paths. I suggest we add additional RoadClass hierarchy levels (8 and 9): 

```
public enum ROAD_CLASS {

        ClassMotorway(0),
        ClassTrunk(1),
        ClassPrimary(2),
        ClassSecondary(3),
        ClassTertiary(4),
        ClassResidential(5),
        ClassUnclassified(6),
        ClassService(7),
        ClassOther(8),                     // other allowing car
        ClassCyclePath(9),               // cycle + ped traffic only
        ClassPedestrianPath(10);     // ped traffic only
}
```

**Comment by hbruch** [comment]
Indeed, we are interested in using SharedStreets to match bike trips. 
As this issue is already some months old: Would you accept a PR adding this feature?

**Comment by quicklywilliam** [comment]
@kpwebb is this the same issue as the one we have been discussing?

**Comment by kpwebb** [comment]
@quicklywilliam it's the same issue but not the same solution -- it turns out that formally inferring bike/ped paths from OSM is more art than science (tl;dr the OSM data simply isn't consistent enough in terms of tagging for a global one size fits all modal tag). The reality is all of the ped/bike paths that are in OSM are mapped in the "other" category already, the challenge is filtering out which ones you want to use in a given location.

We have an alternative solution: use level 8 data, and then allow the user to apply their own filter based on OSM tags in the metadata to add/remove streets from the network. There are some defaults that should get you pretty far (e.g. cycle path tags) but it's likely that you'll also need to special case some ways in the filter, or edit OSM, to make a given location work correctly.

We have a test approach for this and I'll follow up with @michalnaka about next steps, so we can confirm this works for you all.

**Comment by quicklywilliam** [comment]
Thanks @kpwebb
